### PR TITLE
[WIP] Create keycloak BR process

### PR DESCRIPTION
**Pull request by bluzarraga**
For 4.4, develop and test a backup and restore process for keycloak/edb. This is based on the existing process we have for Zen v5's edb. The related files for zen are located in the same directory.

How to test:
- Edit and apply the necessary files (look for any values in `<>` in any files with `keycloak` in the name)
- Can either run velero backup and restore or manually run the scripts from inside the keycloak-backup deployment pod
- verify on same cluster and cluster to cluster that expected data has been restore and functionality has not been impacted

### CODE REVIEW

File: velero/schedule/keycloak-br-script-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            restore
+            success "Restore completed successfully."
+        fi
+    }
+
+    function backup {
+        mkdir -p $BACKUP_DIR/database
+        CNPG_PRIMARY_POD=`oc get cluster.postgresql.k8s.enterprisedb.io keycloak-edb-cluster -o jsonpath="{.status.currentPrimary}" -n $KEYCLOAK_NAMESPACE` && \
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- mkdir -p /run/keycloak_backup && \
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- pg_dump -v --username=postgres --dbname=keycloak -f /run/keycloak_backup/keycloak_db_backup.dump
```
Comment (pgodowski) [inline velero/schedule/keycloak-br-script-cm.yaml:49]: Does this mean we need to increase the PVC side, or use ephemeral storage mount for backup?

File: velero/schedule/keycloak-br-script-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            backup
+            success "Backup completed successfully."
+        else
+            info "Mode is set to restore, beginning restore process."
+            restore
+            success "Restore completed successfully."
+        fi
+    }
+
+    function backup {
```
Comment (pgodowski) [inline velero/schedule/keycloak-br-script-cm.yaml:45]: No need to quiesce the database or enter EDB operator maintenance mode?

File: velero/schedule/keycloak-br-script-cm.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        CNPG_PRIMARY_POD=`oc get cluster.postgresql.k8s.enterprisedb.io keycloak-edb-cluster -o jsonpath="{.status.currentPrimary}"`
+        oc exec $CNPG_PRIMARY_POD -n $KEYCLOAK_NAMESPACE -- mkdir -p /run/keycloak_backup
+        oc cp $BACKUP_DIR/database/keycloak_db_backup.dump $KEYCLOAK_NAMESPACE/$CNPG_PRIMARY_POD:/run/keycloak_backup/keycloak_db_backup.dump
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- psql -U postgres -c "\list" -c "\dn" -c "\du"
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- psql -U postgres -d keycloak -f /run/keycloak_backup/keycloak_db_backup.dump
+        oc -n $KEYCLOAK_NAMESPACE exec -t $CNPG_PRIMARY_POD -c postgres -- psql -U postgres -c "\list" -c "\dn" -c "\du"
+        
+        oc scale deploy rhbk-operator --replicas=$keycloak_replicas -n $KEYCLOAK_NAMESPACE
```
Comment (pgodowski) [inline velero/schedule/keycloak-br-script-cm.yaml:91]: Would keycloak operand STS be scaled back to its initial scale by RHBK operator here?

**Review by pgodowski — CHANGES_REQUESTED** [review]

File: velero/schedule/keycloak-br-script-cm.yaml (hunk 3 above)
Comment (bluzarraga) [reply velero/schedule/keycloak-br-script-cm.yaml:91]: correct

File: velero/schedule/keycloak-br-script-cm.yaml (hunk 2 above)
Comment (bluzarraga) [reply velero/schedule/keycloak-br-script-cm.yaml:45]: The script is based on the steps used by zen 5 (private repo, I'll send via slack) and after discussing with Rahul, it's my understanding that we just need to make sure there are no active users. This is why we have scaled down the keycloak operator. This is something I will need to look more into though to be sure.

**Pull request closed without merging**